packages/passport/sdk/src/zkEvm/transactionHelpers.ts

```diff
@@ -212,3 +219,65 @@ export const prepareAndSignTransaction = async ({
 
   return { signedTransactions, relayerId, nonce };
 };
+
+const buildMetaTransactionForEjection = async (
+  transactionRequest: TransactionRequest,
+): Promise<[MetaTransaction, ...MetaTransaction[]]> => {
+  if (!transactionRequest.to) {
+    throw new JsonRpcError(
+      RpcErrorCode.INVALID_PARAMS,
+      'im_signEjectionTransaction requires a "to" field',
+    );
+  }
+
+  if (!transactionRequest.nonce) {
+    throw new JsonRpcError(
+      RpcErrorCode.INVALID_PARAMS,
+      'im_signEjectionTransaction requires a "nonce" field',
+    );
+  }
+
+  if (!transactionRequest.chainId) {
+    throw new JsonRpcError(
+      RpcErrorCode.INVALID_PARAMS,
+      'im_signEjectionTransaction requires a "chainId" field',
+    );
+  }
+
+  const metaTransaction: MetaTransaction = {
+    to: transactionRequest.to,
+    data: transactionRequest.data,
+    nonce: transactionRequest.nonce,
+    value: transactionRequest.value,
+    revertOnError: true,
```

Comment:
Any reason we do not set the `delegateCall`?

Reply:
unsure myself, what benefit would we get from this? cc @haydenfowler if you have any opinion on this

Reply:
@shirren I believe @matthewmuscat has followed the existing implementation in `eth_sendTransaction` here. As for _why_ `eth_sendTransaction` doesn't set `delegateCall`, I'm not too sure 😅 +1 to Matt's question above